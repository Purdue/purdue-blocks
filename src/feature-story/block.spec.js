jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickRadio,
    clickCheckbox
} from '../test-helpers'

const block = {blockTitle: 'Featured Story', blockName: 'purdue-blocks/feature-story'}

describe( 'ðŸ”¬ Featured Story Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock( block.blockTitle )

        // tests that the block is properly inserted and matches the existing snapshot
        expect( await getEditedPostContent()).toMatchSnapshot()
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
    } )

    describe( 'ðŸ”¬ Block Editor Fields', () => {

        describe('ðŸ”¬ Main content fields', () => {
            test( 'ðŸ”Ž Media Library Button.', async () => {
                await blockStartup(block)
    
                // open media library
                await clickElementByText('button', 'Open Media Library')
    
                // tests that media library does open
                expect(await page.$(`div.media-frame-title`)).not.toBeNull()
            })

            test('ðŸ”Ž Story Header Field', async () => {
                await blockStartup(block)

                const typeString = "Content Header test."
    
                await page.focus('input[placeholder="Content Header..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"header":"${typeString}"`)).toBe(true)
            })

            test('ðŸ”Ž Content Body Field', async () => {
                await blockStartup(block)

                const typeString = "Content main body test."
    
                await page.focus('.block-editor-inner-blocks p[data-type="core/paragraph"]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`<p>${typeString}</p>`)).toBe(true)
            })
        })

        describe('ðŸ”¬ CTA Button fields', () => {
            test('ðŸ”Ž CTA Text Field', async () => {
                await blockStartup(block)

                const typeString = "CTA Text test."
    
                await page.focus('input[placeholder="CTA Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"ctaText":"${typeString}"`)).toBe(true)
            })
            test('ðŸ”Ž CTA URL Field', async () => {
                await blockStartup(block)

                const typeString = "https://www.purdue.edu"
    
                await page.focus('input[placeholder="CTA URL..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"ctaUrl":"${typeString}"`)).toBe(true)
            })
        })
    })

    describe('ðŸ”¬ Side Panel Fields', () => {
        test('ðŸ”Ž Full width option checkbox', async () => {
            await blockStartup(block)
            
            let editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"style":false`)).toBe(false)

            await clickCheckbox('Full width')

            editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"style":false`)).toBe(true)
        })

        test ('ðŸ”Ž Content alignment option radio buttons', async () => {
            await blockStartup(block)

            let editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"contentAlign":"right"`)).toBe(false)

            await clickRadio('Align content', 'right')

            editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"contentAlign":"right"`)).toBe(true)

        })
        
        test( 'ðŸ”Ž Image alt text field.', async () => {
            await blockStartup(block)

            const typeString = "Image alt text."
            
            // focus the text box then type into it with the virtual keyboard
            await clickElementByText('label', 'Featured Image Alt Text')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()

            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
            expect( editedContent).toMatchSnapshot()
        })
        test('ðŸ”Ž Link can be opened in new tab', async () => {
            await blockStartup(block)

            await clickCheckbox('Open link in new tab?')

            const editedContent = await getEditedPostContent()
            expect( editedContent.includes(`"external":true`)).toBe(true)
            expect( await getEditedPostContent() ).toMatchSnapshot()
        })
    })
})