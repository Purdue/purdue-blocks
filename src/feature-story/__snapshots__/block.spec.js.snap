// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Card Block 🔎 Block should be available. 1`] = `
"<!-- wp:purdue-blocks/feature-story -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 CTA Button fields 🔎 CTA Text Field 1`] = `
"<!-- wp:purdue-blocks/feature-story {\\"ctaText\\":\\"CTA Text test.\\"} -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 CTA Button fields 🔎 CTA URL Field 1`] = `
"<!-- wp:purdue-blocks/feature-story {\\"ctaUrl\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 Main content fields 🔎 Content Body Field 1`] = `
"<!-- wp:purdue-blocks/feature-story -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p>Content main body test.</p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Block Editor Fields 🔬 Main content fields 🔎 Story Header Field 1`] = `
"<!-- wp:purdue-blocks/feature-story {\\"header\\":\\"Content Header test.\\"} -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><h2>Content Header test.</h2><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Fields 🔎 Content alignment option radio buttons 1`] = `
"<!-- wp:purdue-blocks/feature-story -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Fields 🔎 Content alignment option radio buttons 2`] = `
"<!-- wp:purdue-blocks/feature-story {\\"contentAlign\\":\\"right\\"} -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image-reversed\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow-reversed\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content content-reversed\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Fields 🔎 Full width option checkbox 1`] = `
"<!-- wp:purdue-blocks/feature-story -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Fields 🔎 Full width option checkbox 2`] = `
"<!-- wp:purdue-blocks/feature-story {\\"style\\":false} -->
<div class=\\"pu-feature-story pu-feature-story__narrow\\"><div class=\\"container pu-feature-story__container\\"><div class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"\\"></div><div class=\\"story-content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;

exports[`🔬 Card Block 🔬 Side Panel Fields 🔎 Image alt text field. 1`] = `
"<!-- wp:purdue-blocks/feature-story {\\"altText\\":\\"Image alt text.\\"} -->
<div class=\\"pu-feature-story\\"><div class=\\"hero is-medium\\"><div class=\\"hero-image\\"><span class=\\"background-image\\" role=\\"img\\" style=\\"background-image:url()\\" aria-label=\\"Image alt text.\\"></span></div><div class=\\"shadow\\"></div><div class=\\"hero-body\\"><div class=\\"container\\"><div class=\\"content\\"><!-- wp:paragraph {\\"placeholder\\":\\"Body content copy\\"} -->
<p></p>
<!-- /wp:paragraph --></div></div></div></div></div>
<!-- /wp:purdue-blocks/feature-story -->"
`;
